### DISCUSSION TITLE
Is it possible to delete a build?

### DISCUSSION THREAD

**Discussion by easherma-truth** [percy/cli#1179] [Support]
Still learning this tool.  I have some builds that I triggered locally, and didn't quite finish as expected. A first build got 203 of my pages, a build inbetween errors out and only got around 60. A subsequent build then got 203 pages successfully. However, the diff is incorrect because it compared against the build missing pages instead of the previous build. 

All that to say, is there a way to remove builds from the list? and/or change the builds being compared and re-run the diffing after the initial snapshot?

**Comment by itsjwala** [comment]
* There is no way to remove a build from the "list".
* what you can do instead is fail the build manually from your CI/local by erroring out your script.
  * Failed build doesn't get counted for checking the diffing.
* How do you fail a build?
  * check for errors in your test suite post each snapshot and exit your script if there are any errors.

---
Additional workaround steps to not upload any snapshots for processing and eventually fail the build.

1. defer uploads 
```yml
percy:
  defer-uploads: true
```

2. Start Percy Local Server https://docs.percy.io/docs/cli-exec#percy-execstart

```sh
npx percy exec:start 
```

3. run your test suite

```sh
# your test suite command 
```

4. kill the Percy local server if your test suite errors out.

```sh
# verify this command once :)
ps -ef | grep percy | grep -v grep |awk '{print $2}' | xargs kill -9 
```

**Note:** Ideal behavior should be a deterministic test suite that will capture all snapshots.

**Reply by easherma-truth** [reply]
Thanks! That helps in some cases. There are other cases where a build may not 'fail' but we'd still want to remove it from the list for comparison. For instance, a snapshot testing layout changes may include some intermediate states that could confuse a designer doing QA. It would be great to be able to 'squash' the diffs to remove those and also make sure the visual diff is run against the version that is most relevant.

**Reply by itsjwala** [reply]
In that case, I'd recommend not creating a build altogether. i.e not spawning the Percy local server by using the `--debug` flag.

example -

```shell
npx percy exec --debug -- # your test command
```

though if your use case still doesn't get solved you can lodge a feature request by reaching out to our support https://browserstack.com/support
